src/basic/capability-util.c

```diff
@@ -223,6 +224,8 @@ int capability_bounding_set_drop(uint64_t keep, bool right_now) {
         r = 0;
 
 finish:
+        if (r == -EPERM && (detect_container() > 0))
+                return 0;
```

Comment:
I really don't like this bit. The thing is that this code is used in various places in systemd, not just in PID 1 (and it should be used everywhere, I think). The logic you added only really makes sense in PID 1 however, afaics...

Moreover I don't really grok what precisely actually fails here... I mean, dropping caps should always be safe: if we have caps we should be able to safely drop them, and if we don't have them then we shouldn't need to drop them and hence not need to generate an error. I don't really grok here what syscall precisely fails in your case. Can you elaborate?

I'd much prefer if we'd do as little container specific checks as necessary here. And in this case it appears that dropping the caps should be fully safe if we have no caps in the first place... hence, please elaborate.

(also, having such "blanket tape over errors a-posteriori" concepts makes all my alarms sound loudly... I would really prefer that if we ignore an error in some condition, it should be as precise as possible, and hence be right to the one syscall where it matters, but not in a global cleanup path for a function, if you follow what I mean)